src/client/datascience/telemetry/interpreterPackages.ts

```diff
@@ -0,0 +1,114 @@
+import { getTelemetrySafeHashedString, getTelemetrySafeVersion } from '../../telemetry/helpers';
+import { Telemetry } from '../constants';
+
+const interestedPackages = new Set(
+    [
+        'ipykernel',
+        'ipython-genutils',
+        'jupyter',
+        'jupyter-client',
+        'jupyter-core',
+        'nbconvert',
+        'nbformat',
+        'notebook',
+        'pyzmq',
+        'pyzmq32',
+        'tornado',
+        'traitlets'
+    ].map((item) => item.toLowerCase())
+);
+
+@injectable()
+export class InterpreterPackages {
+    private static interpreterInformation = new Map<string, Map<string, string>>();
+    private static pendingInterpreterInformation = new Map<string, Promise<void>>();
+    constructor(
+        @inject(IPythonExtensionChecker) private readonly pythonExtensionChecker: IPythonExtensionChecker,
+        @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
+        @inject(IPythonExecutionFactory) private readonly executionFactory: IPythonExecutionFactory
+    ) {}
+    public static getPackageVersions(interpreter: PythonEnvironment): Map<string, string> | undefined {
+        return InterpreterPackages.interpreterInformation.get(interpreter.path);
+    }
+    public trackPackages(interpreterUri: InterpreterUri, ignoreCache?: boolean) {
+        this.trackPackagesInternal(interpreterUri, ignoreCache).catch(noop);
+    }
+    public async trackPackagesInternal(interpreterUri: InterpreterUri, ignoreCache?: boolean) {
+        if (!this.pythonExtensionChecker.isPythonExtensionInstalled) {
+            return;
+        }
+        let interpreter: PythonEnvironment;
+        if (isResource(interpreterUri)) {
+            // Get details of active interpreter for the Uri provided.
+            const activeInterpreter = await this.interpreterService.getActiveInterpreter(interpreterUri);
+            if (!activeInterpreter) {
+                return;
+            }
+            interpreter = activeInterpreter;
+        } else {
+            interpreter = interpreterUri;
+        }
+        this.trackInterpreterPackages(interpreter, ignoreCache).catch(noop);
+    }
+    private async trackInterpreterPackages(interpreter: PythonEnvironment, ignoreCache?: boolean) {
+        const key = interpreter.path;
+        if (InterpreterPackages.pendingInterpreterInformation.has(key) && !ignoreCache) {
+            return;
+        }
+
+        const promise = this.getPackageInformation(interpreter);
+        promise.finally(() => {
+            // If this promise was resolved, then remove it from the pending list.
+            if (InterpreterPackages.pendingInterpreterInformation.get(key) === promise) {
+                InterpreterPackages.pendingInterpreterInformation.delete(key);
+            }
+        });
+        InterpreterPackages.pendingInterpreterInformation.set(key, promise);
+    }
+    @captureTelemetry(Telemetry.AttemptedToLaunchRawKernelWithoutInterpreter)
```

Comment:
This looks like a mistake of some type. This isn't launching the raw kernel.

Reply:
COpy paste, thanks